segmentation/include/pcl/segmentation/impl/extract_clusters.hpp

```diff
@@ -40,14 +40,15 @@
 
 #include <pcl/segmentation/extract_clusters.h>
 
-//////////////////////////////////////////////////////////////////////////////////////////////
-template <typename PointT> void
-pcl::extractEuclideanClusters (const PointCloud<PointT> &cloud,
-                               const typename search::Search<PointT>::Ptr &tree,
-                               float tolerance, std::vector<PointIndices> &clusters,
-                               unsigned int min_pts_per_cluster,
-                               unsigned int max_pts_per_cluster)
-{
+template <typename PointT, typename FunctorT> void
+pcl::extractEuclideanClusters (
+    const PointCloud<PointT> &cloud,
+    const std::vector<int> &indices,
+    FunctorT filter,
+    const typename search::Search<PointT>::Ptr &tree,
+    float tolerance, std::vector<PointIndices> &clusters,
+    unsigned int min_pts_per_cluster,
+    unsigned int max_pts_per_cluster) {
   if (tree->getInputCloud ()->points.size () != cloud.points.size ())
   {
     PCL_ERROR ("[pcl::extractEuclideanClusters] Tree built for a different point cloud dataset (%lu) than the input cloud (%lu)!\n", tree->getInputCloud ()->points.size (), cloud.points.size ());
```

Comment:
Add here all the runtime checks that existed before. I counted at least 3.

Reply:
Line 58, just cast from bool to index_t

Reply:
Done. The normals check can't be added here since the normals point cloud get captured by the lambda and isn't passed explicitly.